### PR TITLE
Check magic resistance against Chain Lighting spell

### PR DESCRIPTION
**Pull request by ihhub**
relates to #1535

### CODE REVIEW

File: src/fheroes2/battle/battle_action.cpp [hunk 1]
```diff
@@ -621,16 +621,38 @@ Battle::TargetsInfo Battle::Arena::GetTargetsForSpells( const HeroBase * hero, c
...
-                trgts.push_back( reslt.size() > 1 ? *Rand::Get( reslt ) : reslt.front() );
+
+                Indexes sortedIds;
+
+                for ( size_t monsterId = 0; monsterId < nearestPosIds.size(); ++monsterId ) {
+                    Unit * target = GetTroopBoard( nearestPosIds[monsterId] );
+                    if ( target != NULL && ( target->GetMagicResist( spell, hero ? hero->GetPower() : 0 ) < 100 ) ) {
+                        sortedIds.push_back( nearestPosIds[monsterId] );
+                    }
+                    ignoredMonster.push_back( nearestPosIds[monsterId] );
```
Comment (idshibanov) [inline src/fheroes2/battle/battle_action.cpp:646]: Why do we need `ignoredMonster`?

Comment (ihhub) [reply src/fheroes2/battle/battle_action.cpp:646]: The problem lies in a situation when we have a creature resistant to spell: Phoenix, for example. If the nearest monster is found to be Phoenix then we need to ignore it and move forward to find a creature which is not resistant. However, I don't really know the correct logic behind the spell. Should we stop adding targets when we reach a magic resistance creature or continue searching for others?

Comment (LeHerosInconnu) [reply src/fheroes2/battle/battle_action.cpp:646]: The Chain Lightning spell always hits four troops, if there are four or more troops on the battlefield.
The Chain Lightning spell never hits the same troop twice.
So, if there are only three troops on the battlefield they will be hit every third and only once.

If a troop resists the spell (eg a Battle Dwarves troop) or is immune to the spell (eg a Phoenixes troop), another troop is chosen.

From my observations, it is always the troop closest to the last troop hit that is hit.
Everything happens as if the troop which resists the spell or is immune to the spell did not exist.

Comment (ihhub) [reply src/fheroes2/battle/battle_action.cpp:646]: This is what I've implemented.

**Pull request merged**